pyccel/parser/semantic.py

```diff
@@ -3239,11 +3241,13 @@ def _visit_Assign(self, expr):
 
                 if is_pointer_i:
                     new_expr = AliasAssign(l, r)
-                    if isinstance(r, (Variable, IndexedElement)):
-                        self._indicate_pointer_target(l, r, expr)
+                    if isinstance(r, FunctionCall):
+                        funcdef = r.funcdef
+                        target_r_idx = funcdef.result_pointer_map[funcdef.results[0]]
```

Comment:
the following code is supported:
```
def f(x: 'int32[:]'):
    x[0] = 5
    y = x
    return y
```
but not this one
```
def f(x: 'int32[:]'):
    x[0] = 5
    y = x
    w = y
    return w
```
Is this left on purpose?

Reply:
I didn't consider this case actually. It could probably be made to work if we think this is useful. What is the error that is raised?

Reply:
Fixed : [27f6231](https://github.com/pyccel/pyccel/pull/1715/commits/27f623127d9ca95f3e34f6efc067dd2a9250a9e6)
And tested : [1943e54](https://github.com/pyccel/pyccel/pull/1715/commits/1943e54ed7b906480280776c7d76bee0e18179b6)